prov/verbs/src/verbs_rma.c

```diff
@@ -276,22 +276,29 @@ fi_ibv_rdm_ep_rma_preinit(void **desc, void **raw_buf, size_t len,
 }
 
 static ssize_t
-fi_ibv_rdm_ep_rma_read(struct fid_ep *ep_fid, void *buf, size_t len,
-		    void *desc, fi_addr_t src_addr,
-		    uint64_t addr, uint64_t key, void *context)
+fi_ibv_rdm_ep_rma_readmsg(struct fid_ep *ep_fid, const struct fi_msg_rma *msg,
+		uint64_t flags)
 {
-	struct fi_ibv_rdm_ep *ep = container_of(ep_fid, struct fi_ibv_rdm_ep,
-						ep_fid);
+	if(msg->iov_count != 1 || msg->rma_iov_count != 1) {
+		assert(0);
+		return -FI_EMSGSIZE;
+	}
+
+	struct fi_ibv_rdm_ep *ep =
```

Comment:
mixed declaration and code is c99 feature.  are we going to force c99 compilation support when not really necessary?